pkg/upgrade/control_plane.go

```diff
@@ -243,7 +251,71 @@ func (u *ControlPlaneUpgrader) etcdClusterHealthCheck(timeout time.Duration) err
 	return err
 }
 
-func (u *ControlPlaneUpgrader) updateMachines(machines *clusterapiv1alpha1.MachineList) error {
+func (u *ControlPlaneUpgrader) updateObjectReference(name string, ref *v1.ObjectReference) (*v1.ObjectReference, *unstructured.Unstructured, error) {
+	u.log.Info("ENTERED UPDATE OBJECT REFERENCE")
+
+	if ref.Namespace == "" {
+		ref.Namespace = "default"
+	}
+	object, err := external.Get(u.ctrlClient, ref, ref.Namespace)
+	if err != nil {
+		u.log.Info("TEST: Error External", "Error", err)
+		return &v1.ObjectReference{}, nil, err
+	}
+
+	object.SetResourceVersion("")
+	object.SetName(name)
+	object.SetOwnerReferences([]metav1.OwnerReference{})
+	unstructured.RemoveNestedField(object.UnstructuredContent(), "spec", "providerID")
```

Comment:
This appears to be used against bootstrap configs, which wouldn't have a `Spec.ProviderID` field.